packages/actor-rdf-resolve-quad-pattern-string-source/test/ActorRdfResolveQuadPatternStringSource-test.ts

```diff
@@ -0,0 +1,203 @@
+      return expect(actor.test({ pattern: <any> null,
+        context: new ActionContext(
+          { [KeysRdfResolveQuadPattern.source.name]:
+            { type: 'stringSource', value: sourceValue, mediaType: sourceMediaType }},
+        ) }))
+        .resolves.toBeTruthy();
+    });
+
+    it('should no test on empty source', () => {
+      const emptySource = '';
+      return expect(actor.test({ pattern: <any> null,
+        context: new ActionContext(
+          { [KeysRdfResolveQuadPattern.source.name]:
+            { type: 'stringSource', value: emptySource, mediaType: sourceMediaType }},
+        ) })).rejects.toBeTruthy();
+    });
+
+    it('should not test on multiple sources', () => {
+      return expect(actor.test(
+        { context: new ActionContext(
+          { [KeysRdfResolveQuadPattern.sources.name]:
+            [{ type: 'stringSource', value: sourceValue, mediaType: sourceMediaType, baseIri: sourceBaseIri },
+              { type: 'stringSource', value: sourceValue, mediaType: sourceMediaType, baseIri: sourceBaseIri },
+            ]},
+        ),
+        pattern: <any> null },
+      ))
+        .rejects.toBeTruthy();
+    });
+
+    it('should not test on the wrong source type', () => {
+      const rdfSource: RDF.Source = { match: () => <any> null };
+      return expect(actor.test({ pattern: <any> null,
+        context: new ActionContext(
+          { [KeysRdfResolveQuadPattern.source.name]: { type: 'foo', value: rdfSource, mediaType: sourceMediaType }},
+        ) })).rejects.toBeTruthy();
+    });
+
+    it('should not test on source with no type', () => {
+      const rdfSource: RDF.Source = { match: () => <any> null };
+      return expect(actor.test({ pattern: <any> null,
+        context: new ActionContext(
+          { [KeysRdfResolveQuadPattern.source.name]: { value: 'stringSource', mediaType: sourceMediaType }},
+        ) })).rejects.toBeTruthy();
+    });
+
+    it('should not test on a source value that is not a string', () => {
+      const rdfSource: RDF.Source = { match: () => <any> null };
+      return expect(actor.test({ pattern: <any> null,
+        context: new ActionContext(
+          { [KeysRdfResolveQuadPattern.source.name]:
+            { type: 'stringSource', value: rdfSource, mediaType: sourceMediaType }},
+        ) })).rejects.toBeTruthy();
+    });
+  });
+
+  describe('The run method of an ActorRdfResolveQuadPatternRdfJsSource instance', () => {
+    let actor: ActorRdfResolveQuadPatternStringSource;
+    const expectedQuads: RDF.Quad[] = [
+      DF.quad(DF.namedNode('ex:a'), DF.namedNode('ex:b'), DF.literal('c')),
+      DF.quad(DF.namedNode('ex:a'), DF.namedNode('ex:d'), DF.literal('e')),
+    ];
+
+    beforeEach(() => {
+      mockMediatorRdfParse = {
+        mediate(_arg: any) {
+          return {
+            handle: {
+              // StreamifyArray has the side effect of comsuming the object hence the clone operation
```

Comment:
I think I rather it this way since I really want the same quads and I feel that it express more the intention and diminish the risk of mistake in case of modification.

Reply:
Ok, but then we need a different way of cloning, because this conversion to raw JSON will cause issues (since object methods are stripped, such as `.equals`).

Reply:
I've added a deep copy library and used it.

Reply:
lodash will still loose the methods though. I would suggest either recreating the quads (simplest solution), or copying them every time using [`DataFactory.fromQuad`](http://rdf.js.org/data-model-spec/#datafactory-interface).